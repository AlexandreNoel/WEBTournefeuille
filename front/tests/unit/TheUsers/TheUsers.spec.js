import { mount } from '@vue/test-utils'
import TheUsers from '@/views/TheUsers.vue'

describe('TheUsers.vue', () => {
  it('should render a table of users', async () => {
    const wrapper = mount(TheUsers, {
      mocks: {
        $api: {
          getUsers () {
            return Promise.resolve(users)
          }
        }
      }
    })

    await wrapper.vm.$nextTick()

    expect(wrapper.html()).toMatchSnapshot()
  })
})

const users = [
  {
    id: 1,
    email: 'test1',
    hash: 'test1',
    pseudo: 'test1',
    role: 'USER'
  },
  {
    id: 2,
    email: 'test2',
    hash: 'test2',
    pseudo: 'test2',
    role: 'USER'
  },
  {
    id: 3,
    email: 'test3',
    hash: 'test3',
    pseudo: 'test3',
    role: 'ADMIN'
  }
]
