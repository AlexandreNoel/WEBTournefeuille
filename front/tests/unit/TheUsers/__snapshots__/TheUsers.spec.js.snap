// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheUsers.vue should render a table of users 1`] = `
<div>
  <div>
    <div class="v-table__overflow">
      <table class="v-datatable v-table theme--light">
        <thead>
          <tr>
            <th role="columnheader" scope="col" aria-label="Id: Sorted ascending. Activate to sort descending." aria-sort="ascending" tabindex="0" class="column sortable active asc text-xs-left">Id<i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">arrow_upward</i></th>
            <th role="columnheader" scope="col" aria-label="Email: Not sorted. Activate to sort ascending." aria-sort="none" tabindex="0" class="column sortable text-xs-left">Email<i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">arrow_upward</i></th>
            <th role="columnheader" scope="col" aria-label="Hash: Not sorted. Activate to sort ascending." aria-sort="none" tabindex="0" class="column sortable text-xs-left">Hash<i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">arrow_upward</i></th>
            <th role="columnheader" scope="col" aria-label="Pseudo: Not sorted. Activate to sort ascending." aria-sort="none" tabindex="0" class="column sortable text-xs-left">Pseudo<i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">arrow_upward</i></th>
            <th role="columnheader" scope="col" aria-label="Role: Not sorted. Activate to sort ascending." aria-sort="none" tabindex="0" class="column sortable text-xs-left">Role<i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">arrow_upward</i></th>
          </tr>
          <tr class="v-datatable__progress">
            <th colspan="5" class="column"></th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>1</td>
            <td>test1</td>
            <td>test1</td>
            <td>test1</td>
            <td>USER</td>
          </tr>
          <tr>
            <td>2</td>
            <td>test2</td>
            <td>test2</td>
            <td>test2</td>
            <td>USER</td>
          </tr>
          <tr>
            <td>3</td>
            <td>test3</td>
            <td>test3</td>
            <td>test3</td>
            <td>ADMIN</td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
